integration_tests/conftest.py

```diff
@@ -1,40 +1,67 @@
+import signal
+import sys
+from typing import List
 import pytest
 import subprocess
 import pathlib
 import os
 import time
 
+def spawn_process(command: List[str]) -> subprocess.Popen[bytes]:
+    if sys.platform.startswith("win32"):
+        command[0] = "python"
```

Comment:
Do you think replacing "python" with "py" might fix the failing windows tests?

Reply:
* I tried replaced `python` with `py` - that did not help
* Apparently `python base_routes.py --dev` does not work in Windows 10 as well so the corresponding test obviously fails, because inside **dev_event_handler.py** we have this:

`self.processes.append(subprocess.Popen(["python3", self.file_name], start_new_session=False))`

* I tried adding a conditional that replaces `python3` with `python` in win32 platform - the dev server starts, but then for some reason it fails to find the modules, including Robyn

Reply:
> Apparently the command python base_routes.py --dev does not work in Windows 10 as well because inside dev_event_handler.py we have this:

This should be the case as we dropped dev server support for windows. But it shouldn't be failing. Can you share the error log please?

Reply:
[windows_test.log](https://github.com/sansyrox/robyn/files/9560577/windows_test.log)

This is the log file obtained

Reply:
> This should be the case as we dropped dev server support for windows. But it shouldn't be failing. Can you share the error log please?

I suspect that `test_dev_index_request()` would still fail anyways, on Windows?

* Trying to run `python base_routes.py --dev` on Windows 10 does not work (in my machine at least), and throws up errors instead
* `test_dev_index_request()` tries to run the aforementioned command, expects a server to get created, and then tries makes a GET request to the server. However, the command seems to fail.

Reply:
Also, `test_global_index_request()` is also failing in Windows. 

```
def test_global_index_request(global_session):
    BASE_URL = "http://0.0.0.0:5000"
    res = requests.get(f"{BASE_URL}")
    assert(os.getenv("ROBYN_URL") == "0.0.0.0")
    assert(res.status_code == 200)
```
* I feel `BASE_URL` should be `http://172.0.0.1` instead? [Reference](url)

>  0.0.0.0 means "all IPv4 addresses on the local machine". It is a meta-address which is non-routable.
>  If you want to access the server locally, i.e. client on the same machine as the server, either use the IP address 127.0.0.1 (loopback Internet protocol) or the equivalent named domain name (localhost)
> r = requests.get('http://127.0.0.1:80')
> r = requests.get('http://localhost:80')

* If this change is made, `test_global_index_request()` would pass in both Windows and Linux. While `requests.get('http://0.0.0.0:8080')` is working fine on Linux, on Windows, it would throw a `requests.exceptions.ConnectionError`.

Reply:
> I suspect that test_dev_index_request() would still fail anyways, on Windows?

Trying to run python base_routes.py --dev on Windows 10 does not work (in my machine at least), and throws up errors instead (traceback included in the previous message)
test_dev_index_request() tries to run the aforementioned command, expects a server to get created, and then tries makes a GET request to the server. However, the command seems to fail.


@guilefoylegaurav , you're correct about this. This needs to be fixed. I will create an issue for this. Thank you!

Reply:
> I feel BASE_URL should be http://127.0.0.1 instead? [Reference](https://github.com/sansyrox/robyn/pull/url)

This should be "0.0.0.0" only as this test is indeed for a session that is available throughout the IP. We have the local ip adress test already availble.